aiidalab_widgets_base/viewers.py

```diff
@@ -554,9 +555,74 @@ def _observe_all_representations(self, change):
         if change["new"]:
             self._all_representations[-1].viewer_class = self
 
+    def _compute_bonds(self, structure, radius=1.0, color="element", povray=False):
+        """Create an array of bonds for the structure."""
+
+        import ase.neighborlist
+
+        bonds = []
+        if len(structure) <= 1:
+            return bonds
+        # The radius is scaled by 0.04 to have a better visual appearance.
+        radius = radius * 0.04
+
+        # The value 1.09 is chosen based on our experience. It is a good compromise between showing too many bonds
+        # and not showing bonds that should be there.
+        cutoff = ase.neighborlist.natural_cutoffs(structure, mult=1.09)
+
+        ii, bond_vectors = ase.neighborlist.neighbor_list(
+            "iD", structure, cutoff, self_interaction=False
+        )
+        nb = len(ii)
+        # bond start position
+        v1 = structure.positions[ii]
+        # middle position
+        v2 = v1 + bond_vectors * 0.5
+
+        # Choose the correct way for computing the cylinder.
+        def povray_cylinder(v1, v2, radius, color):
```

Comment:
I agree using function makes it more clean for me. Can you move it out as `staticmethod` instead of nesting it here? If it is only used in the class, change the name to `_povray_cylinder`. Same for `cylinder` -> `_cylinder` and move it out.

Reply:
Thanks, I changed the inner function into a private helper function.